### ISSUE TITLE
Make colors configurable

### ISSUE THREAD

**Issue by dhaitz** [parrt/dtreeviz#30]
Hi!

Could you make the colors configurable? I like the idea of being colorblind-friendly; unfortunately on some projectors the palette colors are indistinguishable.

Thanks and cheers

**Comment by Ashafix** [comment]
@dhaitz : How should the configuration look like in your opinion? Global or per function call?

As far as I could see from the code, the colors are currently just global constants which can be overwritten, e.g. `dtreeviz.trees.GREY = "#ff0000"` and you get red labels.

Probably the easiests way would be to have a global dictionary which stores the colors and can be overwritten for customization, e.g. "COLORS = {'label': GREY, ', 'title': BLACK}"

@parrt : Any preferences?


```
from sklearn import tree
from sklearn.datasets import load_boston
import dtreeviz.trees

dtreeviz.trees.GREY = "#ff0000"

regr = tree.DecisionTreeRegressor(max_depth=2)
boston = load_boston()
regr.fit(boston.data, boston.target)

viz = dtreeviz.trees.dtreeviz(regr, boston.data, boston.target, target_name='price', feature_names=boston.feature_names)
              
viz.view()
```

**Comment by parrt** [comment]
I think some kind of argument would be better than tweaking the globals.

**Comment by dhaitz** [comment]
@Ashafix , @parrt : Maybe something like this in `dtreeviz/trees.py`?

    GREY = "#ff0000"
    ...
    COLORS = {'label': GREY, ', 'title': BLACK}

    def dtreeviz(..., 
                 colors: dict = None
                 ):

        if colors is None:  # fall back to module-level default colors
            colors = COLORS

**Comment by parrt** [comment]
or just `colors:dict=COLORS`?

**Comment by dhaitz** [comment]
@parrt ah yes, of course this is the simple and straightforward solution.

Another approach would be: 

    if colors is None:
        colors = dict(COLORS, **colors)

which would allow you to overwrite only certain keys in the `colors` argument and use the defaults from `COLORS` for all others.

**Comment by Ashafix** [comment]
Please have a look at https://github.com/Ashafix/dtreeviz/commit/ee02d0ffcb2486f0b1f9eaed5cf53ff778d0798c
and let me know if it's the right direction.

* Each function has an optional parameter `colors` which is a dictionary, each dictionary key can overwrite a default value.
* The colors are stored in `colors.py` allowing easy global modification
* Having a new optional parameter should allow re-use of all previous scripts which use dtreeviz
* `change_colors.py` demonstrates the effects with a few examples
* "tested" on Windows 10 and Ubuntu 16.04

**Comment by parrt** [comment]
@Ashafix Looks pretty good! widespread changes but it's adding a lot of control to the look of the plots.  could you make a PR?  I'm wondering if colors.py should be a separate file or included... thoughts?

**Comment by Ashafix** [comment]
@parrt : Thanks for the quick feedback! Personal opinion, I'd prefer a separate file for the colors/configuration, but not my library.

I will test and document all parameters and then create a PR.

**Comment by parrt** [comment]
sounds great. does extra file cause any package weirdness for users? if invisible, let's factor out into colors.py :)

**Comment by mattharrison** [comment]
While you are on colors, could you do font's as well :) (Ie make them configurable)

**Comment by Ashafix** [comment]
@mattharrison : one PR at a time...
